dlib/test/dnn.cpp

```diff
@@ -1832,6 +1832,12 @@ namespace
             auto res = test_layer(l);
             DLIB_TEST_MSG(res, res);
         }
+        {
+            print_spinner();
+            mish_ l;
+            auto res = test_layer(l);
+            DLIB_TEST_MSG(res, res);
+        }
```

Comment:
On second thought, please also add a test that `cuda::mish()` matches the outputs of `cpu::mish()`.  For example, like is done for `cuda::softmax_all()` in this file as well.

Reply:
Added a simple test.